inference-engine/tests/functional/plugin/cpu/single_layer_tests/depth_to_space.cpp

```diff
@@ -2,56 +2,90 @@
-
-        result << CPUTestsBase::getTestCaseName(cpuParams);
-
-        return result.str();
+        std::tie(inputShapes, inType, mode, blockSize, cpuParams) = obj.param;
+
+        std::ostringstream results;
+        if (!inputShapes.first.empty()) {
+            results << "IS=(";
+            results << CommonTestUtils::partialShape2str(inputShapes.first) << ")_";
+        }
+        results << "TS=";
+        for (const auto& shape : inputShapes.second) {
+            results << CommonTestUtils::vec2str(shape) << "_";
+        }
+        results << "Prc=" << inType << "_";
+        switch (mode) {
+            case DepthToSpace::DepthToSpaceMode::BLOCKS_FIRST:
+                results << "BLOCKS_FIRST_";
+                break;
+            case DepthToSpace::DepthToSpaceMode::DEPTH_FIRST:
+                results << "DEPTH_FIRST_";
+                break;
+            default:
+                throw std::runtime_error("Unsupported DepthToSpaceMode");
+        }
+        results << "BS=" << blockSize << "_";
+        results << CPUTestsBase::getTestCaseName(cpuParams);
+
+        return results.str();
     }
 protected:
     void SetUp() override {
-        LayerTestsDefinitions::depthToSpaceParamsTuple basicParamsSet;
+        DepthToSpaceInputShapes inputShapes;
+        ElementType inType;
+        DepthToSpace::DepthToSpaceMode mode;
+        std::size_t blockSize;
         CPUSpecificParams cpuParams;
-        std::tie(basicParamsSet, cpuParams) = this->GetParam();
+        std::tie(inputShapes, inType, mode, blockSize, cpuParams) = this->GetParam();
 
         std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
-
-        std::vector<size_t> inputShape;
-        DepthToSpace::DepthToSpaceMode mode;
-        std::size_t blockSize;
-        InferenceEngine::Precision netPrecision;
-        std::tie(inputShape, netPrecision, mode, blockSize, targetDevice) = basicParamsSet;
-
-        inPrc = outPrc = netPrecision;
-        selectedType = getPrimitiveType() + "_" + inPrc.name();
-        auto inPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
-        auto params = ngraph::builder::makeParams(inPrc, {inputShape});
-        auto paramOuts = ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
-        auto d2s = ngraph::builder::makeDepthToSpace(paramOuts[0], mode, blockSize);
+        if (selectedType.empty()) {
+            selectedType = getPrimitiveType();
+        }
+        // selectedType = selectedType + "_" + inType;
+        targetDevice = CommonTestUtils::DEVICE_CPU;
+
+        if (!inputShapes.first.empty()) {
+            inputDynamicShapes = inputShapes.first;
+        } else {
+            inputDynamicShapes = { inputShapes.second.front() };
+        }
+        for (size_t i = 0; i < inputShapes.second.size(); i++) {
+            targetStaticShapes.push_back(std::vector<ngraph::Shape>{inputShapes.second[i]});
+        }
```

Comment:
`init_input_shapes({inputShape})` has been introduced in the `SubgraphBaseTest`. Please refer to SoftMax common test class on how to work with input shapes handling.